src/client/render/secondstage.cpp

```diff
@@ -102,21 +103,52 @@ RenderStep *addPostProcessing(RenderPipeline *pipeline, RenderStep *previousStep
 	static const u8 TEXTURE_EXPOSURE_2 = 4;
 	static const u8 TEXTURE_FXAA = 5;
 	static const u8 TEXTURE_VOLUME = 6;
+
+	static const u8 TEXTURE_MSAA_COLOR = 7;
+	static const u8 TEXTURE_MSAA_DEPTH = 8;
+
 	static const u8 TEXTURE_SCALE_DOWN = 10;
 	static const u8 TEXTURE_SCALE_UP = 20;
 
-	// Super-sampling is simply rendering into a larger texture.
-	// Downscaling is done by the final step when rendering to the screen.
-	const std::string antialiasing = g_settings->get("antialiasing");
 	const bool enable_bloom = g_settings->getBool("enable_bloom");
+	const bool enable_volumetric_light = g_settings->getBool("enable_volumetric_lighting") && enable_bloom;
 	const bool enable_auto_exposure = g_settings->getBool("enable_auto_exposure");
+
+	const std::string antialiasing = g_settings->get("antialiasing");
+	const u16 antialiasing_scale = MYMAX(2, g_settings->getU16("fsaa"));
+
+	// This code only deals with MSAA in combination with post-processing. MSAA without
+	// post-processing works via a flag at OpenGL context creation instead.
+
+	// To make MSAA work with post-processing, we need multisample texture support,
+	// which has a higher OpenGL version requirement:
+	// > glTexImage2DMultisample is available only if the GL version is 3.2 or greater.
+	// > ~ https://registry.khronos.org/OpenGL-Refpages/gl4/html/glTexImage2DMultisample.xhtml
+	// Older versions may support it via the GL_ARB_texture_multisample extension.
+
+	// Note: No support for OpenGL ES implemented. That seems to be another can of worms,
+	// different version requirements, different API functions and possibly some extensions.
```

Comment:
looks easy actually
https://docs.gl/es3/glTexStorage2DMultisample

Reply:
hmm yes. I had looked for gles 2, but with gles 3.1+ it's easy.

enabling it results in a huge performance reduction on my phone though.